src/infra/dex/simulator.rs

```diff
@@ -35,12 +36,40 @@ impl Simulator {
     ///
     /// This will return a `None` if the gas simulation is unavailable.
     pub async fn gas(&self, owner: Address, swap: &dex::Swap) -> Result<eth::Gas, Error> {
+        // TODO: use multicall instead
+        // Start simulating the gas for each call in the swap
+        let call_gas_futures = swap
+            .calls
+            .iter()
+            .map(|call| self.call_gas(owner, swap, call));
```

Comment:
Instead of calling `call_gas()` once per transaction and summing the result the `Swapper` helper contract needs to be adjusted to take an array of interactions instead of just 1.
Otherwise the fake settlement executed in `call_gas()` will revert because the `approve` interaction is missing.